src/main/java/baritone/utils/BlockStateInterface.java

```diff
@@ -38,28 +40,31 @@
     private final World world;
     private final WorldData worldData;
 
-
     private Chunk prev = null;
     private CachedRegion prevCached = null;
 
     private static final IBlockState AIR = Blocks.AIR.getDefaultState();
 
+    public BlockStateInterface(IPlayerContext ctx) {
+        this(ctx.world(), (WorldData) ctx.worldData());
+    }
+
     public BlockStateInterface(World world, WorldData worldData) {
         this.worldData = worldData;
         this.world = world;
     }
 
-    public static Block getBlock(BlockPos pos) { // won't be called from the pathing thread because the pathing thread doesn't make a single blockpos pog
-        return get(pos).getBlock();
+    public static Block getBlock(IPlayerContext ctx, BlockPos pos) { // won't be called from the pathing thread because the pathing thread doesn't make a single blockpos pog
+        return get(ctx, pos).getBlock();
     }
 
-    public static IBlockState get(BlockPos pos) {
-        return new CalculationContext().get(pos); // immense iq
+    public static IBlockState get(IPlayerContext ctx, BlockPos pos) {
+        return new BlockStateInterface(ctx).get0(pos.getX(), pos.getY(), pos.getZ()); // immense iq
```

Comment:
Okay I'll give you that it's better to just create a BlockStateInterface not a CalculationContext but still, not the best. Idk.

Reply:
nOt the best? we're technically I'm removing like 3 method calls.

<details>
Owned with facts and logic.
</details>

Reply:
Yes it's better but it's not ideal, that's what i said.

<details>
rekt
</details>

Reply:
FUU how else could I improve this?

Reply:
Eh it's good enough IMO.